reward/src/lib.rs

```diff
@@ -0,0 +1,461 @@
+    let note = repo.find_note(Some(NOTES_REF), *selected_commit)?;
+    log::debug!("Selected note: {:?}", note.id());
+
+    let note = match note.message() {
+        Some(proof) => proof,
+        None => bail!("Not able to obtain commit message"),
+    };
+
+    let proof: Proof = serde_json::from_str(note)?;
+    log::debug!("Retrieved Proof: {:?}", proof);
+
+    let puzzle = Puzzle {
+        org: proof.org,
+        contributor: proof.contributor,
+        commit: proof.commit,
+        project: proof.project,
+        uri: proof.uri.clone(),
+    };
+
+    Ok((puzzle, proof))
+}
+
+pub fn select_commit(commits: &[Oid]) -> anyhow::Result<Oid> {
+    let selection = Select::with_theme(&ColorfulTheme::default())
+        .items(commits)
+        .with_prompt("Claimable Commits")
+        .interact_on_opt(&Term::stderr())?;
+
+    let index = match selection {
+        Some(index) => index,
+        None => bail!("User did not select any commit"),
+    };
+
+    Ok(commits[index])
+}
+
+pub fn get_eligible_commits(
+    signer_address: Address,
+    repo: &Repository,
+) -> anyhow::Result<Vec<Oid>> {
+    let mut commits: Vec<Oid> = Vec::new();
+
+    for note in repo.notes(Some(NOTES_REF))? {
+        let oids = note?;
+        let note = repo.find_note(Some(NOTES_REF), oids.1)?;
+        let message = note.message().unwrap();
+        let t: Proof = serde_json::from_str(message)?;
+        if signer_address == t.contributor {
+            commits.push(oids.1);
+        }
+    }
+
+    Ok(commits)
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    /// No wallet specified.
+    #[error("Missing Argument: {0}")]
+    ArgMissing(String),
+    /// Not able to retrieve block .
+    #[error("not able to retrieve block")]
+    NoBlock,
+    /// Not able to retrieve block hash .
+    #[error("not able to retrieve block hash")]
+    NoBlockHash,
+    /// Not able to retrieve commit.
+    #[error("not able to retrieve commit")]
+    CommitNotExisting,
+    /// Not able to sign message
+    #[error("not able to sign message")]
+    SignFailure,
+    /// Not able to sign message
+    #[error("not able to sign message")]
+    SerializeFailure,
+    /// Not valid commit summary
+    #[error("{0} not valid")]
+    NotValidEncoding(String),
+    /// ETH signature failed
+    #[error("eth signature failed")]
+    ETHSigFailed,
+    /// GPG signature failed
+    #[error("{0}")]
+    GPGSigFailed(String),
+    /// Ledger Signer Error
+    #[error(transparent)]
+    Ledger(#[from] ethers::prelude::LedgerError),
+    /// Ethers Wallet Error
+    #[error(transparent)]
+    Wallet(#[from] ethers::prelude::WalletError),
+}
+
+/// The options allowed to be provided to the CLI
+#[derive(Debug, Clone)]
+pub struct Options {
+    /// Address of org.
+    pub org: Option<Address>,
+    /// Address of contributor
+    pub contributor: Option<Address>,
+    /// Repo path
+    pub repo: Option<PathBuf>,
+    /// Project id.
+    pub project: Option<String>,
```

Comment:
Are `repo` and `project` ever optional? How about `org`?

Reply:
I left everything optional, since `--create` needs almost all the arguments, `--claim` needs only `--repo`.